src/synapse/azext_synapse/custom.py

```diff
@@ -153,8 +165,9 @@ def create_spark_pool(cmd, client, resource_group_name, workspace_name, spark_po
         big_data_pool_info.auto_pause = AutoPauseProperties(enabled=auto_pause_enabled,
                                                             delay_in_minutes=delay_in_minutes)
 
-    if library_requirements_filename or library_requirements_content:
-        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_filename,
+    if library_requirements_file:
+        library_requirements_content = read_file_content(library_requirements_file)
+        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_file,
                                                                       content=library_requirements_content)
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, workspace_name, spark_pool_name,
                        big_data_pool_info, force)

```

Comment:
This comment is for `az synapse spark pool update`. We can switch auto-scale and auto-pause through portal, but we cannot do the same thing with the command. So I suggest to change to use SDK method `create_or_update`

Reply:
Also we should allow user to modify node count.

Reply:
> Also we should allow user to modify node count.

fixed. Now `az synapse spark pool update` can update node count , auto scale, auto pause etc.